libr/debug/debug.c

```diff
@@ -1239,11 +1244,13 @@ R_API int r_debug_continue_kill(RDebug *dbg, int sig) {
 			goto repeat;
 		}
 
+#if !__linux__
```

Comment:
uhm, can you confirm in which other OS is not the same?

Reply:
Sure, I will go find out why this line is necessary. Imho, the `dbg->tid` returned by `r_debug_continue` does not necessarily mean which thread is stopped and it should rely on `r_debug_wait` to tell which thread changes its states, so these lines are not necessary on Linux.

This is just a quick way to not break other debugger plugins which rely on these lines. For Linux native debugger, it relies on `r_debug_wait` to change the `dbg->tid` because the `dbg->tid` returned by `continue` which will always be the same as before `continue`.

Reply:
After reviewing this agian, `r_debug_select` should be called here actually. However, some code in the debug_native plugin call this directly to switch to a thread/process and get information from it, which is messing with `RDebug`'s state and should be avoided.